cmd/traffic/cmd/manager/internal/mutator/agent_injector.go

```diff
@@ -0,0 +1,228 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	admission "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/datawire/ambassador/pkg/kates"
+	"github.com/datawire/dlib/dlog"
+	"github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager/managerutil"
+	"github.com/telepresenceio/telepresence/v2/pkg/install"
+)
+
+var podResource = metav1.GroupVersionResource{Version: "v1", Group: "", Resource: "pods"}
+var findMatchingService = install.FindMatchingService
+
+func agentInjector(ctx context.Context, req *admission.AdmissionRequest) ([]patchOperation, error) {
+	// This handler should only get called on Pod objects as per the MutatingWebhookConfiguration in the YAML file.
+	// Pod objects are immutable, hence we only care about the CREATE event.
+	// Applying patches to Pods instead of Deployments means we don't have side effects on
+	// user-managed Deployments and Services. It also means we don't have to manage update flows
+	// such as removing or updating the sidecar in Deployments objects... a new Pod just gets created instead!
+
+	// If (for whatever reason) this handler is invoked on an object different than a Pod,
+	// issue a log message and let the object request pass through.
+	if req.Resource != podResource {
+		dlog.Debugf(ctx, "expect resource to be %s, got %s; skipping", podResource, req.Resource)
+		return nil, nil
+	}
+
+	// Parse the Pod object.
+	raw := req.Object.Raw
+	pod := corev1.Pod{}
+	if _, _, err := universalDeserializer.Decode(raw, nil, &pod); err != nil {
+		return nil, fmt.Errorf("could not deserialize pod object: %v", err)
+	}
+
+	podNamespace := pod.Namespace
+	if podNamespace == "" {
+		// It is very probable the pod was not yet assigned a namespace,
+		// in which case we should use the AdmissionRequest namespace.
+		podNamespace = req.Namespace
+	}
+	podName := pod.Name
+	if podName == "" {
+		// It is very probable the pod was not yet assigned a name,
+		// in which case we should use the metadata generated name.
+		podName = pod.ObjectMeta.GenerateName
+	}
```

Comment:
That... can't be right?  The Pod name is `GenerateName` plus a random suffix.
```
$ kubectl get pod -o yaml verylargejavaservice-6b696cb89f-blfdd |y2j|jq '{name: .metadata.name, generateName: .metadata.generateName}'
{
  "name":         "verylargejavaservice-6b696cb89f-blfdd",
  "generateName": "verylargejavaservice-6b696cb89f-"
}
```

Reply:
Copy pasta from old impl. I'll remote it.